test/arrays.js

```diff
@@ -583,4 +583,11 @@
     assert.deepEqual(_.chunk([10, 20, 30, 40, 50, 60, 70], 2), [[10, 20], [30, 40], [50, 60], [70]], 'chunk into parts of less then current array length elements');
     assert.deepEqual(_.chunk([10, 20, 30, 40, 50, 60, 70], 3), [[10, 20, 30], [40, 50, 60], [70]], 'chunk into parts of less then current array length elements');
   });
+
+  QUnit.test('remove', function(assert) {
+    assert.deepEqual(_.remove([1, 2, 3, 4], [1, 3, 4]), function(obj) {return(obj == 2)});
```

Comment:
This line seems to have been written with the following intuition: if you have the elements 1, 2, 3, 4 and you remove the elements 1, 3, 4, you'll be left with only element 2. While this intuition is right, this is not what you should be testing, and the way you're trying to test it also doesn't work.

You shouldn't be testing this, because the intent of `_.remove` is to take as its second argument a *criterion* (`predicate`) by which to decide whether an element should be removed, not a *list of values* to be removed. A function that accepts a list instead of a criterion as its second argument would be the modifying version of `_.difference` instead of the modifying version of `_.reject`. As-is (with your current implementation) the argument `[1, 3, 4]` will in fact be interpreted as a predicate, i.e., it will be passed to `_.iteratee` and be converted to a function equivalent to `function(x) { return x[1][3][4]; }`. I'm sure this is not at all what you meant; I think you meant `function(x) { return x == 1 || x == 3 || x == 4; }`.

The test also doesn't work, because `assert.deepEqual()` takes two arguments and then recursively compares them by value. You're passing it two arguments: the result of `_.remove([1, 2, 3, 4], [1, 3, 4])` and a function. Unless you expect `_.remove` to return a function, this assertion is always going to fail. You should pass the expected value as the second argument to `assert.deepEqual`, not a function that performs a comparison.

There is also something missing. You should not only compare the result of `_.remove` to an expected value, but also verify that this result is the same array as the one that you passed as the first argument. After all, the intent of `_.remove` is that it modifies the array in place.